tests/e2e/operator/operator_test.go

```diff
@@ -384,3 +366,59 @@ func LogDebugInfo() {
 	}
 	GinkgoWriter.Println("Logs from istiod pod: \n", logs)
 }
+
+// key returns the client.ObjectKey for the given name and namespace. If no namespace is provided, it returns a key cluster scoped
+func key(name string, namespace ...string) client.ObjectKey {
+	if len(namespace) > 1 {
+		panic("you can only provide one namespace")
+	} else if len(namespace) == 1 {
+		return client.ObjectKey{Name: name, Namespace: namespace[0]}
+	}
+	return client.ObjectKey{Name: name}
+}
+
+// getObject returns the object with the given key
+func getObject(ctx context.Context, cl client.Client, key client.ObjectKey, obj client.Object) (client.Object, error) {
+	err := cl.Get(ctx, key, obj)
+	return obj, err
+}
+
+// checkNamespaceEmpty checks if the given namespace is empty
+func checkNamespaceEmpty(ctx SpecContext, ns string) {
```

Comment:
I think that technically all checks should be performed in a single `Eventually`, because we expect the namespace to be empty at one moment in time. Currently, this function would succeed even if some pods pop up when the function is checking services.

If it's too hard to use a single `Eventually`, I'm okay with leaving this as is.